src/menuFactory.js

```diff
@@ -186,7 +186,7 @@ export default (styles) => {
 
     componentWillReceiveProps(nextProps) {
       // Allow open state to be controlled by props.
-      if (nextProps.isOpen !== this.state.isOpen) {
+      if (nextProps.isOpen !== this.props.isOpen && nextProps.isOpen !== this.state.isOpen) {
```

Comment:
Not really, what you propose is what was already there before this PR :) if you revert the change here, the test will fail. What we really want to do is to find out if the `isOpen` prop was changed before comparing the new `isOpen` prop with `isOpen` in state (because the only reason that open state controlled by props was changed is that the `isOpen`prop was changed).

Maybe it wasn't completely clear from PR description, but consider this:
```jsx
const MenuComponent = React.createClass({
     getInitialState() {
         return { collapsed: true };
     },
     toggle(e) {
          e.preventDefault();

          this.setState({
              collapsed: !this.state.collapsed
          });
     },
     render() {
         return (
             <Menu>
                 <a onClick={this.toggle}>Toggle</a>

                 {/* collapsible div follows */}
                 <div className={this.state.collapsed ? 'collapsed' : 'expanded'}>
                     { /* content */ }
                </div>
                {/* ... bunch of other links outside of the collapsible div */} 
             </Menu>
         );
     }
 });
```

We are not setting `isOpen` prop on `<Menu />` component. So `isOpen` prop is always false, but by clicking on the burger icon, `<Menu />`'s `isOpen` in state is changing from false to true.

When I click on `Toggle` link inside the `<Menu />` component, parent component's inner state will change (from `collapsed: true` to `collapsed: false`), causing the whole parent component to rerender itself so that the collapsible div's className is set to `extended`. That means the `<Menu />` component will receive new props. But those props are the same as it was before.

So if we keep the condition in `componentWillReceiveProps` as it was before (`(nextProps.isOpen !== this.state.isOpen)`), it will be evaluated to true (because `false !== true`) so the `toggleMenu` method will be triggered, causing the `<Menu />` to close. But what I really wanted is only to expand the collapsed div inside of `<Menu />` component.

And this is not what we want as `<Menu />`s `isOpen` prop was not changed. By introducing addition check to  `componentWillReceiveProps` condition (`nextProps.isOpen !== this.props.isOpen`) we ensure that the only reason why the `isOpen` prop should affect `isOpen` state is only when the `isOpen` prop was changed.

Hope it makes sense :)